node-api/server/server.go

```diff
@@ -22,56 +22,95 @@ package server
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/berachain/beacon-kit/chain"
 	"github.com/berachain/beacon-kit/log"
 	"github.com/berachain/beacon-kit/log/noop"
+	"github.com/berachain/beacon-kit/node-api/backend"
 	"github.com/berachain/beacon-kit/node-api/handlers"
+	beaconapi "github.com/berachain/beacon-kit/node-api/handlers/beacon"
+	builderapi "github.com/berachain/beacon-kit/node-api/handlers/builder"
+	configapi "github.com/berachain/beacon-kit/node-api/handlers/config"
+	debugapi "github.com/berachain/beacon-kit/node-api/handlers/debug"
+	eventsapi "github.com/berachain/beacon-kit/node-api/handlers/events"
+	nodeapi "github.com/berachain/beacon-kit/node-api/handlers/node"
+	proofapi "github.com/berachain/beacon-kit/node-api/handlers/proof"
+	"github.com/berachain/beacon-kit/node-api/middleware"
+	"github.com/berachain/beacon-kit/node-core/components/storage"
+	"github.com/berachain/beacon-kit/node-core/types"
+	cmtcfg "github.com/cometbft/cometbft/config"
 )
 
 // Server is the API Server service.
 type Server struct {
-	engine Engine
-	config Config
-	logger log.Logger
+	config     Config
+	logger     log.Logger
+	middleware *middleware.Middleware
 }
 
 // New initializes a new API Server with the given config, engine, and logger.
 // It will inject a noop logger into the API handlers and engine if logging is
 // disabled.
 func New(
 	config Config,
-	engine Engine,
 	logger log.Logger,
-	handlers ...handlers.Handlers,
-) *Server {
+
+	// attributes to build handlers backend
+	storageBackend *storage.Backend,
+	cs chain.Spec,
+	cmtCfg *cmtcfg.Config,
+	consensusService types.ConsensusService,
+) (*Server, error) {
 	apiLogger := logger
 	if !config.Logging {
 		apiLogger = noop.NewLogger[log.Logger]()
 	}
+
+	mware := middleware.NewDefaultMiddleware()
+
+	// instantiate handlers and register their routes in the middleware
+	b, err := backend.New(storageBackend, cs, cmtCfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed instantiating handlers backend: %w", err)
+	}
+	b.AttachQueryBackend(consensusService)
```

Comment:
this used to be done, post-hoc, in node builder. I think it's cleaner to do it here, and clearly signal in one place all the relevant dependencies.